### ISSUE TITLE
Merging the fix of CVE-2022-36087 into tag v3.2.1

### ISSUE THREAD

**Issue by noren95** [oauthlib/oauthlib#837]
Hello,

I would like to get some information/verification regarding the fix for CVE-2022-36087.
According to the release notes and the related pull request generated by the Security Advisory, a patch is available in version [3.2.1](https://github.com/oauthlib/oauthlib/releases/tag/v3.2.1).
However, when reviewing the vulnerable file `uri_validate.py`, it seems to remain similar to the previous, affected version v3.2.0. In other words, it seems that the fix was merged into master only. When looking at the file through v3.2.1, it was changed recently on May 10, 2021.

Can you please confirm that version 3.2.1 is officially patched, even though it might be not reflected in the repository? If it does, please share any additional references in addition to the available ones. 

References:
https://github.com/oauthlib/oauthlib/blob/v3.2.1/oauthlib/uri_validate.py
https://github.com/oauthlib/oauthlib/commit/2e40b412c844ecc4673c3fa3f72181f228bdbacd
https://github.com/advisories/GHSA-3pgj-pg6c-r5p7

Kind Regards,
Noa

**Comment by benknoble** [comment]
Agreed; other reports list 3.2.1 as still affected, even though the NIST report does not.

**Comment by jacwalte** [comment]
Still seeing this issue.  Are there any updates?

**Comment by auvipy** [comment]
git log shows https://github.com/oauthlib/oauthlib/commit/2e40b412c844ecc4673c3fa3f72181f228bdbacd was merged

**Comment by auvipy** [comment]
release notes https://github.com/oauthlib/oauthlib/releases/tag/v3.2.1

**Comment by noren95** [comment]
Apparently, it was merged but into 'master' only.
The vulnerable line still exists on v3.2.1:
https://github.com/oauthlib/oauthlib/blob/v3.2.1/oauthlib/uri_validate.py#L69

**Comment by auvipy** [comment]
@JonathanHuot can you chime in please?

**Comment by JonathanHuot** [comment]
I have released 3.2.2 with patched file. I've tried to update CVE from within github, but it seems not reflecting the info anywhere. I will look into creating a new one if required later.

Thanks for the report! It was a mistake during previous release preparation.

**Comment by noren95** [comment]
Hi @JonathanHuot 
Thank you for the updates!
